src/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+# Source files
+file(GLOB_RECURSE SOURCES_SC2LADDERSERVER "*.cpp" "*.h")
+
+# Include directories
+include_directories(SYSTEM
+    ${PROJECT_SOURCE_DIR}/s2client-api/include
+    ${PROJECT_SOURCE_DIR}/s2client-api/contrib/protobuf/src
+    ${PROJECT_BINARY_DIR}/s2client-api/generated
+    ${PROJECT_SOURCE_DIR}/rapidjson
+    ${PROJECT_SOURCE_DIR}/curl/include/curl
+)
+
+# Link directories
+link_directories(${PROJECT_BINARY_DIR}/s2client-api/bin)
+link_directories(${PROJECT_SOURCE_DIR}/curl/lib)
```

Comment:
No need to specify lib curl on Linux/OS X. We can't link with DLLs anyway.

Reply:
To be honest I usually try to avoid submodules. From my point of view it is better to install lib curl from a site (we can even add a make command/target for it) instead of submodules.

Reply:
I don't think I understand sorry.
Do you mean I should check the platform before linking the /curl/lib directory?

Reply:
Ah, i see!
I had anticipated the binaries for other platforms being in there.
Is this wrong? If so, I will add the check in, as you have suggested.

Reply:
>No idea about OS X, however, but I'll add the check in as you suggested.

For OS X it is harder. One can use macports, homebrew, bring a library using its installer or use source code/precompiled libs. The real problem is the difference in the libraries. Currently I can’t compile the ladder manager with OS X native version of libcurl. Thats why I suggested to allow exclude it and the related code from compilation.